src/main/java/org/mule/extension/runtime/ExpirationPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.mule.extension.runtime;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A policy around when should a given instances be considered elegible for expiration.
+ * <p/>
+ * Notice that this contract is not directly tied to the instance to be expired itself. It's tied
```

Comment:
I don't get this paragraph. What do you mean with "It's tied to the concept of idle based expiration"?